File: leetcode/0053-maximum-subarray/solution.py
class Solution(object):
    def maxSubArray(self, nums):
        max_sum = curr = nums[0]
        for i in nums[1:]:
            curr = max(i,curr+i)
            max_sum = max(curr,max_sum)

        return max_sum
        """
        :type nums: List[int]
        :rtype: int
        """
        

File: leetcode/0053-maximum-subarray/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let maxSum = nums[0]
    let curr = 0
    for(let num of nums){
        curr+=num
        maxSum = Math.max(curr,maxSum)
        if(curr<0){
            curr = 0
        }

    }
    return maxSum
    
};
